pkg/controller/controller_instance_test.go

```diff
@@ -215,7 +215,63 @@ func TestReconcileInstanceNonExistentServicePlan(t *testing.T) {
 	}
 }
 
-func TestReconcileInstanceWithParameters(t *testing.T) {
+func TestReconcileInstanceWithInvalidParameters(t *testing.T) {
+	fakeKubeClient, fakeCatalogClient, fakeBrokerClient, testController, sharedInformers := newTestController(t, noFakeActions())
+
+	sharedInformers.Brokers().Informer().GetStore().Add(getTestBroker())
+	sharedInformers.ServiceClasses().Informer().GetStore().Add(getTestServiceClass())
+
+	instance := getTestInstance()
+
+	// Secret with invalid JSON
+	invalidSecret := &v1.Secret{
+		Data: map[string][]byte{
+			testSecretKey: []byte("invalid json"),
```

Comment:
I'm having flashbacks to my days of working on Docker build where we tried to automagically determine if the CMD was a string/shell-cmd vs json array and we couldn't tell because things that started with '[' could be either.    :-)   The real trick was determining when to generate an error because you can't tell if a malformed json object was bad json vs just a string that happens to almost be valid json.    I think being explicit (and not trying to guess) is the safest.